File: Data Structures & Algorithms/max-water-container/submission-0.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let left=0;
        let right = heights.length-1
        let max=0
        while(left<right){
            let higher = Math.min(heights[left], heights[right])
            let area = higher *(right-left)
            max= Math.max(area, max)
            if(heights[left]<heights[right]){
                left++;
            }else{
                right--
            }
        }
        return max
    }
}

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        cost.append(0)
        dp = [-1 for x in range(len(cost))]
        def dfs(i):
            if i < 0:
                return 0
            
            if dp[i] !=-1 : return dp[i]
            right = float("-inf")
            left = cost[i] + dfs(i-1)
            if i>-1:
                right = cost[i] + dfs(i-2)
            dp[i] = min(left,right)
            return dp[i]

            
        return dfs(len(cost)-1)